MoreLinq/FallbackIfEmpty.cs

```diff
@@ -163,14 +163,12 @@ public static IEnumerable<T> FallbackIfEmpty<T>(this IEnumerable<T> source, IEnu
         {
             if (source is ICollection<T> collection)
             {
-                if (collection.Count == 0)
-                {
-                    return Fallback();
-                }
-                else
-                {
-                    return collection;
-                }
+                return collection.Count == 0 ? Fallback() : collection;
+            }
+
+            if (source is IReadOnlyCollection<T> readOnlyCollection)
```

Comment:
Like in #383, this can be consolidated with the above as both lead to the same results (`Fallback()` or `source`) predicated only on the collection count.